### DISCUSSION TITLE
Novel Roadmap

### DISCUSSION THREAD

**Discussion by andrewdoro** [steven-tey/novel#273] [Announcements]
## Maintainer Update 
 🥳 Hello, I'm Andrew, and I will be joining this project as a maintainer. I am very thankfull to @steven-tey for allowing me to work on this. I always wanted to be more involved in open-source work.
 
 I wanted to initiate a discussion about Novel's future plans and outline what I believe is the best course of action. I am currently using Novel in a production project and plan to continue its application in the future. I'm particularly impressed by the simplicity of a Notion-like editor, and my aim is to enhance this library's utility as a versatile, drop-in editor for any project. 

**My first goal**: While I'm aware of the numerous outstanding issues and PRs, my initial focus will be on merging PR #136. This step is crucial as it will simplify subsequent additions I plan to make. Although I cannot provide an exact completion date, I anticipate that it should not take longer than a week, given that most of the code is still relevant. This PR aims to introduce unstyled components and will adopt a **Radix-like** structure, emphasizing heavily on **composability**. 

**Short tem goals:** ⟹
- [x] #136 Finish and merge this PR 
- [ ] Clean up all the remaining PR's and issues
- [x] ~~Create a `novel/headless` and `novel/tailwind` for the npm package~~
- [x] Multiple package entries for extensions, plugins ...

**Long term goals: ⟹ ⟹ ⟹**
- [ ] Add new building blocks to the editor such as [tldraw](https://tldraw.dev/)
- [ ] ~~Investigate using [plate](https://platejs.org/) instead of [tiptap](https://tiptap.dev/) for the underline logic~~
- [ ] Implement a similar CLI to shadcn/ui for adding features to the editor
- [x] A proper documentation website using `Nextra` or `Starlight` - WIP using mintlify
- [ ] Multiple demos and use cases for the editor #254 

The list I provided represents my initial thoughts. However, I welcome your opinions and ideas. If you have any suggestions for features or different plans for this project, please feel free to share them. 🥳🥳🥳

<img alt="nah_id_win" src="https://github.com/steven-tey/novel/assets/17547035/7b146f04-0010-4511-82a6-13d7cdea21e4"  height="200" />

**Comment by steven-tey** [comment]
Incredibly excited to have you join as a maintainer, Andrew! Can't wait to see where you'll take this project 🤩

**Comment by rishi-raj-jain** [comment]
Congratulations, Andrew! 🚀

Wish you the best!

**Comment by amiranvarov** [comment]
Nice! Agree, providing headless option would be quote handy. Thank you for stepping up :)

**Comment by yassine-chrtk** [comment]
Congratulations @andrewdoro ! Wish you the best. 
Please ping me on the documentation website project. I'd be happy to help there as much as I can!

**Comment by okanji** [comment]
Great!

**Comment by faltawy** [comment]
Congrats andrew 🎊

**Comment by maks-ivanov** [comment]
Really looking forward to #136 - awesome work!

**Comment by mamlzy** [comment]
Can't wait to see Novel getting better!

**Comment by andrewdoro** [comment]
## Canary package [0.2.0-canary.0](https://www.npmjs.com/package/novel/v/0.2.0-canary.0)
Released a canary package with the headless approach [0.2.0-canary.0](https://www.npmjs.com/package/novel/v/0.2.0-canary.0)
See here an example implementation until I have the documentation up:
 https://github.com/steven-tey/novel/blob/main/apps/web/app/page.tsx
 ```tsx
import {
  EditorBubble,
  EditorBubbleItem,
  EditorCommand,
  EditorCommandItem,
  EditorContent,
  EditorRoot,
} from "novel";

export default () => (
  <EditorRoot>
    <EditorContent>
      <EditorCommand>
        <EditorCommandItem />
        <EditorCommandItem />
        <EditorCommandItem />
      </EditorCommand>
      <EditorBubble>
        <EditorBubbleItem />
        <EditorBubbleItem />
        <EditorBubbleItem />
      </EditorBubble>
    </EditorContent>
  </EditorRoot>
);
```

**Reply by chinnuu05** [reply]
Great job with this new release! It looks like AI features have been moved inside the new "Ask AI" 

I'm not sure about others but for my use case streaming the response straight into the editor makes the most sense. I am interested in helping solve [#58 ](https://github.com/steven-tey/novel/issues/58) if you are open to this @andrewdoro

**Reply by andrewdoro** [reply]
Hey @chinnuu05 really cool if you can help with that. Currently I am still looking over the AI feature, I wanted to move it out of the core library and have it more like a custom addition. I would detail more about this in the docs.

**Reply by chinnuu05** [reply]
@andrewdoro think I've got it working, [take a look](https://i.imgur.com/R9d3Yu7.gif). Definitely need some sort of loading UI because responses can take a while. 

Also I think being able to toggle the new "Ask AI" menu by typing ++ would be great

**Comment by andrewdoro** [comment]
### WIP docs available here 
https://novel.sh/docs/introduction

**Reply by mamlzy** [reply]
So cool! thank you so much!

**Comment by jerriclynsjohn** [comment]
Really curious to know your analysis on Plate vs Tiptap for this project, I can see that @andrewdoro you have crossed it off the list. Is there any insights that you could possibly share?

**Reply by chinnuu05** [reply]
I think Tiptap is just more mature and it has a rich library of extensions too. Most of Novel is built on those extensions, so switching would kind of a pain.
